app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
+      find_user_from_basic_auth
+    else
+      # Authenticate the user internally
+      perform_basic_auth
+    end
+
+    render_authorization_required unless @authenticated_user.present?
```

Comment:
I also noticed that in the original PR, is `@authenticated_user` ever set in the new code?

The old code contained an explicit `@authenticated_user = xyz`, I can't find it in the new code and also don't see that it is set as a side effect.